src/json/motion.rs

```diff
@@ -33,6 +34,152 @@ pub struct Meta {
     pub size_user_data: usize,
 }
 
+/// Point.
+#[derive(Copy, Clone, Debug, Serialize, Deserialize)]
+pub struct SegmentPoint {
+    /// Time.
+    pub time: f32,
+    /// Value.
+    pub value: f32,
+}
+
+/// Segment.
+#[derive(Copy, Clone, Debug)]
+pub enum Segment {
+    /// Linear.
+    Linear(SegmentPoint, SegmentPoint),
+    /// Bezier curve.
+    Bezier([SegmentPoint; 4]),
+    /// Stepped.
+    Stepped(SegmentPoint, f32),
+    /// Inverse stepped.
+    InverseStepped(f32, SegmentPoint),
+}
+
+mod segment_parser {
+    use crate::json::motion::{Segment, SegmentPoint};
+    use serde::{
+        self,
+        de::{self, SeqAccess, Visitor},
+        Deserializer, Serializer,
+    };
+
+    struct SegmentVisitor;
+
+    impl<'de> Visitor<'de> for SegmentVisitor {
+        type Value = Vec<Segment>;
+        fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+            formatter.write_str("sequence of intergers / numbers")
```

Comment:
you made a small typo here with `intergers`

Reply:
Fixed in 177e920.